File: Models/FriendProfileModel.cs
using System;

namespace ChatApp.Models
{
    public class FriendProfileModel
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }

        public string UserName { get; set; }   

        public string Email { get; set; }

        public string ImageUrl { get; set; }
        public string Designation { get; set; }

        public string? LastMessage { get; set; } = null;

        public DateTime? LastMessageTimeStamp { get; set; } = null;

        public int UnreadMessageCount { get; set; }
    }
}

File: MainApp/src/app/core/models/friend-profile-model.ts
import { DesignationModel } from './designation-model';

export interface FriendProfileModel {
  firstName: string;
  lastName: string;
  userName: string;
  email: string;
  imageUrl: string;
  designation: DesignationModel;
  lastMessage?: string;
  lastMessageTimeStamp?: Date;
  unreadMessageCount?: number;
}
